v1/transform.go

```diff
@@ -0,0 +1,344 @@
+	parameterNames := make([]string, len(resourceOperations))
+	for i, ro := range resourceOperations {
+		parameterNames[i] = ro.DeviceResource
+	}
+	action := v1Models.Action{
+		Path: fmt.Sprintf("/api/v1/device/{deviceId}/%s", cmdName),
+		Responses: []v1Models.Response{
+			{
+				Code:        strconv.Itoa(http.StatusNoContent),
+				Description: fmt.Sprintf("Issue the Put command %s", cmdName),
+			}, {
+				Code:        strconv.Itoa(http.StatusInternalServerError),
+				Description: "internal server error",
+			},
+		},
+	}
+	return v1Models.Put{
+		Action:         action,
+		ParameterNames: parameterNames,
+	}
+}
+
+// TransformToV2DeviceProfile transform v1 profile to v2
+func TransformToV2DeviceProfile(profile v1Models.DeviceProfile) models.DeviceProfile {
+	dto := models.DeviceProfile{
+		Description:  profile.Description,
+		Name:         profile.Name,
+		Manufacturer: profile.Manufacturer,
+		Model:        profile.Model,
+		Labels:       profile.Labels,
+	}
+	dto.DeviceResources = toV2DeviceResources(profile)
+	dto.DeviceCommands = toV2DeviceCommands(profile.DeviceCommands)
+	return dto
+}
+
+func toV2DeviceResources(profile v1Models.DeviceProfile) []models.DeviceResource {
+	resources := make([]models.DeviceResource, len(profile.DeviceResources))
+	for i, r := range profile.DeviceResources {
+		resources[i] = models.DeviceResource{
+			Description: r.Description,
+			Name:        r.Name,
+			IsHidden:    isResourceHidden(r.Name, profile),
+			Tags:        toV2Tags(r.Tags),
+			Properties: models.ResourceProperties{
+				ValueType:    strings.Title(strings.ToLower(r.Properties.Value.Type)),
+				ReadWrite:    r.Properties.Value.ReadWrite,
+				Units:        r.Properties.Units.DefaultValue,
+				Minimum:      r.Properties.Value.Minimum,
+				Maximum:      r.Properties.Value.Maximum,
+				DefaultValue: r.Properties.Value.DefaultValue,
+				Mask:         r.Properties.Value.Mask,
+				Shift:        r.Properties.Value.Shift,
+				Scale:        r.Properties.Value.Scale,
+				Offset:       r.Properties.Value.Offset,
+				Base:         r.Properties.Value.Base,
+				Assertion:    r.Properties.Value.Assertion,
+				MediaType:    r.Properties.Value.MediaType,
+			},
+			Attributes: toV2Attributes(r.Attributes),
+		}
+	}
+
+	return resources
+}
+
+func isResourceHidden(resourceName string, profile v1Models.DeviceProfile) bool {
+	// Check whether the resource exists in the CoreCommands, if exists, the resource is not hidden.
+	for _, coreCommand := range profile.CoreCommands {
+		for _, res := range coreCommand.Get.Responses {
+			isContains := contains(res.ExpectedValues, resourceName)
+			if isContains {
+				return false
+			}
+		}
+		isContains := contains(coreCommand.Put.ParameterNames, resourceName)
+		if isContains {
+			return false
+		}
+	}
+	// Check whether the resource exists in the DeviceCommands, if exists, the resource is not hidden.
+	for _, deviceCommand := range profile.DeviceCommands {
```

Comment:
DeviceCommand means nothing to the isHidden of DeviceResource

Reply:
Removed this code snippet.